torch/_decomp/decompositions.py

```diff
@@ -785,6 +785,25 @@ def diagonal_backward(
     return torch.diagonal_scatter(grad_input, grad_output, offset, dim1, dim2)
 
 
+@register_decomposition(aten.block_diag)
+@out_wrapper()
+def block_diag(tensors: List[Tensor]):
+    ncols = sum([tensor.shape[1] for tensor in tensors])
+    device = tensors[0].device
```

Comment:
This should have more input validation to check the tensors are on the same device and to promote scalar and 1-d tensors to 2-dimensions.